### PR TITLE
update readme for TGL

### PR DESCRIPTION
**Pull request by XinfengZhang**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,23 +100,23 @@ Media driver supports two build types as below
...
-| HEVC 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/E/Es</u><br><i>D/E<i> |
-| HEVC 10bit 422 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/Es</u><br><i>D<i> |
-| HEVC 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
+| CODEC | Build Types | BDW | SKL | BXT/APL | KBL | CFL | WHL | CML | ICL* | TGL |
```
Comment (dvrogozh) [inline README.md]: Not related to TGL, but spotted in this review. Pay attention at "ICL*" comment below under the table:
> ICL encoding is pending on i915 support on upstream, for more information, please check [Known Issues and Limitations #5](https://github.com/intel/media-driver/blob/master/README.md#known-issues-and-limitations).

I think this is outdated and required patches landed. We need to fix that.

File: README.md [hunk 2]
```diff
@@ -171,12 +172,14 @@ or later, which officially supported C++11.
 3. HuC firmware is needed for AVC/HEVC/VP9 low power encoding bitrate control, including CBR, VBR, etc. As of now, HuC firmware support is disabled in Linux kernels by default. Please, refer to i915 kernel mode driver documentation to learn how to enable it. Mind that HuC firmware support presents in the following kernels for the specified platforms:
    * APL/KBL: starting from kernel 4.11
    * CFL: starting from kernel 4.15
+   * ICL: starting from kernel 4.8
```
Comment (dvrogozh) [inline README.md]: You are saying that ICL appeared _before_ CFL?! That's kind of impossible from my perpective. Refer to: https://github.com/Intel-Media-SDK/MediaSDK/wiki/Intel-Graphics-Support-in-Linux-Kernels#all-kernels. ICL support in general appeared only in 4.17 kernel under alpha support.

File: README.md [hunk 3]
```diff
@@ -171,12 +172,10 @@ or later, which officially supported C++11.
 3. HuC firmware is needed for AVC/HEVC/VP9 low power encoding bitrate control, including CBR, VBR, etc. As of now, HuC firmware support is disabled in Linux kernels by default. Please, refer to i915 kernel mode driver documentation to learn how to enable it. Mind that HuC firmware support presents in the following kernels for the specified platforms:
    * APL/KBL: starting from kernel 4.11
    * CFL: starting from kernel 4.15
+   * ICL: starting from kernel 5.2
+   
+4. TGL: preliminary support and may not be fully functional on 5.4.2, need config i915.force_probe=* and no Huc, Guc loaded. [drm-tip](https://cgit.freedesktop.org/drm-tip) should work with more support and bug fixes.
```
Comment (dvrogozh) [inline README.md]: Suggestions:
1. Give range of kernels where TGL support appears and note that it is yet preliminary. See my variant:
> TGL: preliminary support and may not be fully functional. It requires Linux kernel 5.4+ loaded with i915.force_probe=* parameter to be tried. [drm-tip](https://cgit.freedesktop.org/drm-tip) should work with more support and bug fixes.

2. Don't mix up with HUC and GUC here. Just drop the text.

File: README.md [hunk 4]
```diff
@@ -83,6 +83,7 @@ export LIBVA_DRIVER_NAME=iHD
 * WHL (Whiskey Lake)
 * CML (Comet Lake)
 * ICL (Ice Lake)
+* TGL (Tiger Lake)
```
Comment (dvrogozh) [inline README.md]: Do we support EHL in the driver? If yes, do we need to mention it separately in the whole readme?

File: README.md [hunk 5]
```diff
@@ -171,12 +172,10 @@ or later, which officially supported C++11.
 3. HuC firmware is needed for AVC/HEVC/VP9 low power encoding bitrate control, including CBR, VBR, etc. As of now, HuC firmware support is disabled in Linux kernels by default. Please, refer to i915 kernel mode driver documentation to learn how to enable it. Mind that HuC firmware support presents in the following kernels for the specified platforms:
    * APL/KBL: starting from kernel 4.11
    * CFL: starting from kernel 4.15
+   * ICL: starting from kernel 5.2
```
Comment (dvrogozh) [inline README.md]: As of now FW for these platforms did not appear even in the latest 5.4.2: EHL, TGL:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c?h=v5.4.2#n42

But it did appear on in drm-tip:
https://cgit.freedesktop.org/drm-tip/tree/drivers/gpu/drm/i915/gt/uc/intel_uc_fw.c#n46

Hence I suggest to add this to the readme:
> * EHL, TGL: [drm-tip](https://cgit.freedesktop.org/drm-tip), missed in <= 5.4.2 kernels

File: README.md (hunk 3 above)
Comment (XinfengZhang) [reply README.md]: agree, will change

File: README.md (hunk 4 above)
Comment (XinfengZhang) [reply README.md]: I would like to merge KBL/CFL/WHL/CML to one column KBL+ and all derivative platform should be just a explanation

File: README.md (hunk 5 above)
Comment (XinfengZhang) [reply README.md]: agree

File: README.md (hunk 4 above)
Comment (dvrogozh) [reply README.md]: Yep, this makes sense for the tables below. They become too wide. Though I would list everything in this list here one by one to make sure people understand we support derivative platforms.

File: README.md [hunk 6]
```diff
@@ -78,11 +78,10 @@ export LIBVA_DRIVER_NAME=iHD
 * BDW (Broadwell)
 * SKL (Skylake)
 * BXT (Broxton) / APL (Apollo Lake)
-* KBL (Kaby Lake)
-* CFL (Coffee Lake)
-* WHL (Whiskey Lake)
-* CML (Comet Lake)
+* KBL+ (KBL/Kaby Lake; CFL/Coffe Lake; WHL/Whiskey Lake; CML/Comet Lake; AML/Amber Lake)
```
Comment (dvrogozh) [inline README.md]: Can you, please, drop +? Someone can read that as everything above KBL gen9 is supported. If you wanted to have abbreviation which will mean a family, then maybe use KBL* or KBLx?

File: README.md [hunk 7]
```diff
@@ -99,29 +98,29 @@ Media driver supports two build types as below
...
-| HEVC 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/E/Es</u><br><i>D/E<i> |
-| HEVC 10bit 422 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/Es</u><br><i>D<i> |
-| HEVC 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
+| CODEC | Build Types | BDW | SKL | BXT/APL | KBL+ | ICL | TGL* |
```
Comment (dvrogozh) [inline README.md]: See above, + is confusing

File: README.md [hunk 8]
```diff
@@ -171,12 +170,11 @@ or later, which officially supported C++11.
 3. HuC firmware is needed for AVC/HEVC/VP9 low power encoding bitrate control, including CBR, VBR, etc. As of now, HuC firmware support is disabled in Linux kernels by default. Please, refer to i915 kernel mode driver documentation to learn how to enable it. Mind that HuC firmware support presents in the following kernels for the specified platforms:
    * APL/KBL: starting from kernel 4.11
    * CFL: starting from kernel 4.15
+   * ICL: starting from kernel 5.2
+   * EHL/JSL: [drm-tip](https://cgit.freedesktop.org/drm-tip), missed in <=5.4.2 kernels
```
Comment (dvrogozh) [inline README.md]: you miss TGL and change <=5.4.2 to 5.4.3 :)

File: README.md [hunk 9]
```diff
@@ -78,11 +78,10 @@ export LIBVA_DRIVER_NAME=iHD
 * BDW (Broadwell)
 * SKL (Skylake)
 * BXT (Broxton) / APL (Apollo Lake)
-* KBL (Kaby Lake)
-* CFL (Coffee Lake)
-* WHL (Whiskey Lake)
-* CML (Comet Lake)
+* KBL* (KBL/Kaby Lake; CFL/Coffe Lake; WHL/Whiskey Lake; CML/Comet Lake; AML/Amber Lake)
 * ICL (Ice Lake)
+* JSL (Jasper Lake)/EHL (Elkhart Lake)
```
Comment (dvrogozh) [inline README.md]: Don't JSL/EHL belong to gen11 family? Maybe you need to list them in ICL line? /this gives issue for the table below/

File: README.md [hunk 10]
```diff
@@ -99,29 +98,29 @@ Media driver supports two build types as below
...
-| HEVC 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/Es</u><br><i>D<i> | <u>D/E/Es</u><br><i>D/E<i> |
-| HEVC 10bit 422 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/Es</u><br><i>D<i> |
-| HEVC 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 8bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D</u><br><i>D<i> | <u>D/E</u><br><i>D/E<i> |
-| VP9 10bit 444 | <u>Full Feature</u><br><i>Free Kernel</i> |  |  |  |  |  |  |  | <u>D/E</u><br><i>D/E<i> |
+| CODEC | Build Types | BDW | SKL | BXT/APL | KBL* | ICL | TGL* |
```
Comment (dvrogozh) [inline README.md]: This gives a question: which features are exposed for JSL and EHL? And using * for both KBL and TGL gives confusion. Maybe write in this way:
| CODEC | Build Types | BDW | SKL | BXT/APL | KBL* | ICL | TGL** |

\* Marks SKU family. See "Supported Platforms" for the list of SKUs which belong for particular family
\** TGL encoding is pending on i915 support on upstream, for more information, please check [Known Issues and Limitations #4, #5](https://github.com/intel/media-driver/blob/master/README.md#known-issues-and-limitations).

File: docs/media_features.md [hunk 11]
```diff
@@ -81,4 +81,6 @@
 |SKL/BXT/KBL/CFL/WHL/CML| Input  |  Y   |  Y   |  Y   |  Y   |  Y   |      |      |      |      |
 |                       | Output |  Y   |  Y   |  Y   |      |  Y   |      |      |      |      |
 |      ICL              | Input  |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |
+|                       | Output |  Y   |  Y   |  Y   |  Y   |  Y   |      |  Y   |  Y   |  Y   |
+|      TGL              | Input  |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |  Y   |
```
Comment (dvrogozh) [inline docs/media_features.md]: Where JSL and EHL belong to?

Comment (XinfengZhang) [reply docs/media_features.md]: @FurongZhang please help to check the vpp feature list in the table especially of EHL/JSL

File: README.md [hunk 12]
```diff
@@ -78,11 +78,10 @@ export LIBVA_DRIVER_NAME=iHD
 * BDW (Broadwell)
 * SKL (Skylake)
 * BXT (Broxton) / APL (Apollo Lake)
-* KBL (Kaby Lake)
-* CFL (Coffee Lake)
-* WHL (Whiskey Lake)
-* CML (Comet Lake)
+* KBLx (KBL/Kaby Lake; CFL/Coffe Lake; WHL/Whiskey Lake; CML/Comet Lake; AML/Amber Lake)
```
Comment (dvrogozh) [inline README.md]: yep, this will do

**Comment by dvrogozh** [comment]
There is a conflict in docs/media_features.md - please, rebase. Also, please, squash commits.

**Comment by XinfengZhang** [comment]
I rebased and squashed it , will merge it

**Pull request merged**